extensions/amp-subscriptions/0.1/amp-subscriptions.js

```diff
@@ -294,7 +295,11 @@ export class SubscriptionService {
             this.fetchEntitlements_(subscriptionPlatform);
           }
       );
-      this.startAuthorizationFlow_();
+
+      // Delegates the platform selection and activation call.
+      const doPlatformSelection = !isStoryDocument(this.ampdoc_.win.document);
```

Comment:
Critically! The first argument cannot be `win.document`. This method must be derived from the ampdoc and its root node.

Reply:
I wrote a clone of `src/dom.waitForBody` method, that waits for `body.firstChildElement` instead. This method uses `win.document` though, like `waitForBody`, please lmk if there's something I'm missing here